### PR TITLE
Update setup.py with allowed numpy version

### LINKED ISSUE #69: pyHyp doesn't work with numpy 1.24
## Description

Self explanatory. We are trying to update the "latest" docker images to numpy 1.24 (https://github.com/mdolab/docker/pull/208) and getting the error below in all the pyHyp tests. This error looks pretty much identical to https://github.com/mdolab/adflow/issues/256 so the solution will likely also be similar

### Steps to reproduce issue

1. Install numpy v1.24
2. Run pyHyp tests

### Current behavior

```text
2023-01-17T17:05:27.9879234Z (mpi) ./test_all_examples.py:TestExamples.test_sphere ... FAIL (00:00:0.94, 315 MB)
2023-01-17T17:05:27.9880090Z Traceback (most recent call last):
2023-01-17T17:05:27.9881809Z   File "/home/***/.pyenv/versions/3.10.9/lib/python3.10/site-packages/testflo/test.py", line 418, in _try_call
2023-01-17T17:05:27.9886788Z     func()
2023-01-17T17:05:27.9887811Z   File "/home/***/repos/pyhyp/tests/./test_all_examples.py", line 98, in test_sphere
2023-01-17T17:05:27.9888961Z     from examples.sphere.runSphere import volumeFile, commonOptions
2023-01-17T17:05:27.9889972Z   File "/home/***/repos/pyhyp/tests/../examples/sphere/runSphere.py", line 55, in <module>
2023-01-17T17:05:27.9891336Z     hyp = pyHypMulti(options=options, commonOptions=commonOptions, skipList=skipList)
2023-01-17T17:05:27.9892759Z   File "/home/***/repos/pyhyp/tests/../pyhyp/pyHyp.py", line 225, in __init__
2023-01-17T17:05:27.9893807Z     hypGrid = pyHyp(comm, options, debug)
2023-01-17T17:05:27.9895165Z   File "/home/***/repos/pyhyp/tests/../pyhyp/pyHyp.py", line 424, in __init__
2023-01-17T17:05:27.9895976Z     self._setOptions()
2023-01-17T17:05:27.9896977Z   File "/home/***/repos/pyhyp/tests/../pyhyp/pyHyp.py", line 763, in _setOptions
2023-01-17T17:05:27.9898101Z     self.hyp.hypinput.sourcestrengthfile = self._expandString("")
2023-01-17T17:05:27.9899552Z ValueError: 0-th dimension must be 512 but got 0 (not defined).
```

### Expected behavior
All tests pass

### PR DESCRIPTION
**Pull request by eirikurj**
## Purpose
Numpy just released 1.24.3, so issues described in #69 should be addressed. The `setup.py` now excludes the problematic 1.24 versions, but should permit everything above 1.16. If we know of other versions, we should add them also.

Closes #69 

## Expected time until merged
No rush.

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
     ),
     package_data={"pyhyp": ["*.so"]},
     install_requires=[
-        "numpy>=1.16,<1.24",
+        "numpy>=1.16,!=1.24,!=1.24.1,!=1.24.2",
```
Comment (sseraj) [inline setup.py:34]: Do we need `!=1.24`? I would think `!=1.24.1,!=1.24.2` is enough?

Comment (eirikurj) [reply setup.py:34]: No, we need it since version 1.24.0 (and some RCs) would be matched otherwise.

**Review by sseraj — APPROVED** [review]
@eirikurj Please override the checks and merge

**Comment by sseraj** [comment]
Well apparently I can merge this

**Pull request merged**